src/com/phonebookservice/controller/AbstractController.java

```diff
@@ -0,0 +1,78 @@
+package com.phonebookservice.controller;
+
+import java.util.HashMap;
+import java.util.List;
+
+import com.phonebookservice.model.Model;
+import com.phonebookservice.server.IDataAccessAdapter;
+import com.phonebookservice.util.EnumQueryParams;
+import com.phonebookservice.util.QueryParam;
+
+/**
+ * abstract controller.
+ *
+ * @author Marwa Saleh
+ * @param <T> model
+ */
+public abstract class AbstractController<T extends Model> {
+    private final IDataAccessAdapter database;
```

Comment:
Is it right ??! I think we discussed it before. 

How will u know the result of type of T ?!!

Reply:
you didn't fix it. You didn't answer to me about this problem

Reply:
I know u made getDatabase returns of type <T>. You should at least answer me here.

Reply:
I know if we already discussed it on the phone then no need to add it here !
I did not fix it cause we said it did not give any error when we added it in get and want to know why